### PR TITLE
Buffs the Blastcannon

### PR DESCRIPTION
**Pull request by TemporalOroboros**
If you don't know what the blastcannon is (Honestly, who would? I've never seen anyone else pick it; probably because it's garbage.) the blastcannon is a 14 TC traitor weapon available to scientists. It uses TTVs for ammunition and, in exchange for sacrificing all of the splash radius of the TTV, basically generates an immovable rod that carves a trench through the station. At least, that's what it's intended to do.

As it turns out, the blastcannon has two problems. Firstly, it could never do the looping immovable rod thing that the wiki and most game sources claim it could because it's range has always been capped to 150. Secondly, it has literally been hugboxed, and as carries little of the destructive power of the weapon used to create it. As such it is generally considered to be an utter waste of TC by the people who actually know that it exists.

This PR attempts to change this. Firstly, it unhugboxes the blastcannon which lets it use most of the destructive power of the bomb used to fire it. Secondly, it makes the blastcannon's range scale with the explosive power of the bomb used to create it. Hopefully it will actually see some use after this change! It's no fun expending more than half of your TC on an item that sounds fun, but turns out to be almost complete trash.

I also fixed a minor runtime where the blastcannon would hand SSexplosions a null when it hit its max range. 

## Changelog
:cl:
balance: <i>You press your ear to the barrel and hear a faint rumba beat...</i> The blastcannon has been buffed!
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
No idea what the implications of this are but sounds fun

File: code/modules/projectiles/guns/misc/blastcannon.dm [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
 	fire_sound = 'sound/weapons/blastcannon.ogg'
 	item_flags = NONE
 	clumsy_check = FALSE
 	randomspread = FALSE
+	/// The icon state used when this is loaded with a bomb.
+	var/icon_state_loaded = "loaded_blastcannon"
 
-	var/hugbox = TRUE
+	/// Whether or not
+	var/hugbox = FALSE
```
Comment (Ghommie) [inline code/modules/projectiles/guns/misc/blastcannon.dm:19]: I doubt anyone will code an hugboxed version of this gun anytime soon. Feel free to remove the variable and its code.

File: code/modules/projectiles/guns/misc/blastcannon.dm [hunk 2]
```diff
@@ -1,26 +1,39 @@
+/**
+ *
+ */
```
Comment (Ghommie) [inline code/modules/projectiles/guns/misc/blastcannon.dm:3]: Empty comment.

File: code/modules/projectiles/guns/misc/blastcannon.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/modules/projectiles/guns/misc/blastcannon.dm]: Done

File: code/modules/projectiles/guns/misc/blastcannon.dm (hunk 2 above)
Comment (TemporalOroboros) [reply code/modules/projectiles/guns/misc/blastcannon.dm]: Un-emptied

**Comment by cacogen** [comment]
Blast cannon is a vague and generic name you should use Cuban Pete's name recognition to sell this puppy

**Comment by TemporalOroboros** [comment]
I mean, I would. But for one I don't think we are supposed to add player references, and for two I can't think of a good name for it based off of Cuban Pete. Blastcannon might be gneric, but at least it rolls of the tongue.

**Comment by Winterous** [comment]
I've only used it a couple of times to snipe people through security cameras (god I love that feature), and I love restoring it to glory.

**Comment by cacogen** [comment]
>ultra rare Pete's blast cannon!

I can't think of anything either.

**Comment by TheVekter** [comment]
I recall these hitting like a fucking truck before, though that was a while back. What's the realistic killing power of it after the buffs?

**Comment by Winterous** [comment]
Against mobs it looks like there's no change, but against walls I think it's changed from some snowflakey turf destruction to raw explosion effects.
That said due to the increased range cap, you'll get stronger impacts from a greater distance, so in practice you'll be able to gib people from further away.

**Comment by TemporalOroboros** [comment]
@TheVekter Winterous basically got it. They gib at close range, crit at medium range, and knock people around at long range. Before it would _always_ crit regardless of the range. The maximum range has been extended from 150 turfs so it can actually do the looping rod thing if the z-levels wraps.

**Comment by Cyprex** [comment]
> I mean, I would. But for one I don't think we are supposed to add player references.
Cuban Pete is referenced throughout the game.

File: code/modules/projectiles/guns/misc/blastcannon.dm [hunk 3]
```diff
@@ -47,54 +61,58 @@
...
-		bomb = T
-		name = "blast cannon"
-		desc = "A makeshift device used to concentrate a bomb's blast energy to a narrow wave."
-		update_icon()
-		return TRUE
-	return ..()
+	. = ..()
+	icon_state = bomb ? icon_state_loaded : initial(icon_state)
+
+/obj/item/gun/blastcannon/attackby(obj/item/transfer_valve/_bomb, mob/user)
```
Comment (Jared-Fogle) [inline code/modules/projectiles/guns/misc/blastcannon.dm:67]: I don't like using `_name` to refer to things that aren't unused, change this to `new_bomb` or similar.

Comment (Jared-Fogle) [reply code/modules/projectiles/guns/misc/blastcannon.dm:67]: Or just name it `bomb` and do `src.bomb = bomb` at the end, I'm quite liking that style.

**Pull request merged**